operator/gefyra/bridge_mount/duplicate.py

```diff
@@ -0,0 +1,336 @@
+        )
+
+    @property
+    def _original_pods(self) -> V1PodList:
+        return self._get_pods_workload(
+            name=self.target,
+            namespace=self.namespace,
+            workload_type="deployment",
+        )
+
+    # TODO this also exists in the client, should be moved to a shared location
+    def _get_pods_workload(
+        self, name: str, namespace: str, workload_type: str
+    ) -> V1PodList:
+        API_EXCEPTION_MSG = "Exception when calling Kubernetes API: {}"
+        NOT_FOUND_MSG = f"{workload_type.capitalize()} not found."
+        try:
+            if workload_type == "deployment":
+                workload = app.read_namespaced_deployment(
+                    name=name, namespace=namespace
+                )
+            elif workload_type == "statefulset":
+                workload = app.read_namespaced_stateful_set(
+                    name=name, namespace=namespace
+                )
+        except ApiException as e:
+            if e.status == 404:
+                # TODO better exception typing here
+                raise Exception(NOT_FOUND_MSG)
+            raise RuntimeError(API_EXCEPTION_MSG.format(e))
+
+        # use workloads metadata uuid for owner references with field selector to get pods
+        v1_label_selector = workload.spec.selector.match_labels
+
+        label_selector = ",".join(
+            [f"{key}={value}" for key, value in v1_label_selector.items()]
+        )
+
+        if not label_selector:
+            # TODO better exception typing here
+            raise Exception(f"No label selector set for {workload_type} - {name}.")
+        pods = core_v1_api.list_namespaced_pod(
+            namespace=namespace, label_selector=label_selector
+        )
+        return pods
+
+    def _get_svc_fqdn(self) -> str:
+        svc_name = self._get_duplication_svc_name()
+        return f"{svc_name}.{self.namespace}.svc.cluster.local"
+
+    def _set_carrier_upstream(self, pod: V1Pod, container: V1Container) -> None:
+        timeout = 30
+        waiting_pod = core_v1_api.read_namespaced_pod(
+            name=pod.metadata.name, namespace=self.namespace
+        )
+        # wait for pod to be ready
+        while (
+            not self.pod_ready_and_healthy(waiting_pod, self.container) and timeout > 0
+        ):
+            sleep(1)
+            timeout -= 1
+            waiting_pod = core_v1_api.read_namespaced_pod(
+                name=pod.metadata.name, namespace=self.namespace
+            )
+        if timeout == 0:
+            raise RuntimeError(f"Pod {pod.metadata.name} did not become ready in time")
+        carrier = Carrier2(
+            configuration=self.configuration,
+            target_namespace=self.namespace,
+            target_pod=pod.metadata.name,
+            target_container=self.container,
+            logger=self.logger,
+        )
+        # TODO what about multiple ports?
+        carrier.carrier_config.port = container.ports[0].container_port
```

Comment:
@Schille I noticed the carrier config.yaml only allows for one port. Is this correct? Therefore we currently select only one port here (but how would we know which one is correct?